### ISSUE TITLE
Critical: overflow in RoaringIntIterator:advanceIfNeeded

### ISSUE THREAD

**Issue by jandam** [RoaringBitmap/RoaringBitmap#110]
Version: 0.6.17

java.lang.ArrayIndexOutOfBoundsException: -2147483648
	at org.roaringbitmap.RoaringArray.getContainerAtIndex(RoaringArray.java:339)
	at org.roaringbitmap.RoaringBitmap$RoaringIntIterator.nextContainer(RoaringBitmap.java:88)
	at org.roaringbitmap.RoaringBitmap$RoaringIntIterator.advanceIfNeeded(RoaringBitmap.java:97)

this.pos can be greater than RoaringBitmap.this.highLowContainer.size() when last containers are empty.
  pos is incremented in while cycle but nextContainer do nothing.

Workaround:
  Use standard RoaringIntIterator::hasNext/next

Suggested fix:
```
   public void advanceIfNeeded(int minval) {
            while((('\uffff' & this.hs >>> 16) < ('\uffff' & minval >>> 16)) && hasNext()) { // FIX for this issue
                ++this.pos;
                this.nextContainer();
            }

            if(this.hasNext()) {
                this.iter.advanceIfNeeded(Util.lowbits(minval));
                if(!this.iter.hasNext()) {
                    ++this.pos;
                    this.nextContainer();
                }
            }

        }

```

**Comment by lemire** [comment]
Verified. New release is in production.

**Comment by lemire** [comment]
Hopefully fixed in 0.6.18. Reopen if you disagree. Please consider adding a failing test if your choose to re-open.

**Comment by jandam** [comment]
Thank you very much for fast release. Fix is working for me.
